### DISCUSSION TITLE
Documentation Request: Google OAuth on Flutter

### DISCUSSION THREAD

**Discussion by alvinsga** [pocketbase/pocketbase#1986] [Q&A]
Documentation on how to integrate OAuth on Flutter and persist the login session does not currently exist

**Comment by ganigeorgiev** [comment]
Docs contributions are welcomed. For now I postponed the initially planned docs updates [due to other more major tasks](https://github.com/orgs/pocketbase/projects/2).

The current OAuth2 flow should be the same as the web example - https://pocketbase.io/docs/authentication/#web-oauth2-integration. The main difference is that instead of a redirect, you have to handle a deep link.
For flutter, you can also check this discussion - https://github.com/pocketbase/pocketbase/discussions/442#discussioncomment-4160820.

_Keep in mind that the OAuth2 flow will be simplified in the near future as part of #55._

---

For Dart SDK persistence, you can check https://github.com/pocketbase/dart-sdk/issues/13.
As mentioned in the related issue, there are too many options and for now this is left to the developers to handle on their own. A simple example using shared_preferences could be found in https://github.com/pocketbase/pocketbase/discussions/1887#discussioncomment-5057297.
